src/PhpParser/Node/BetterNodeFinder.php

```diff
@@ -544,6 +535,44 @@ public function resolveCurrentStatement(Node $node): ?Stmt
         return null;
     }
 
+    private function resolveNextNode(Node $node, ?Node $parentNode): ?Node
+    {
+        if (! $parentNode instanceof Node) {
+            $newStmts = $this->resolveNewStmts($parentNode);
+            return $this->resolveNodeFromFile($newStmts, $node, false);
+        }
+
+        if ($node instanceof Stmt) {
+            if (! $parentNode instanceof StmtsAwareInterface) {
+                return null;
+            }
+
+            if ($parentNode->stmts === null) {
+                return null;
+            }
+
+            // todo: use +1 key once all next node attribute reference and NodeConnectingVisitor removed
+            // left with add SlimNodeConnectingVisitor for only lookup parent
+            return $node->getAttribute(AttributeKey::NEXT_NODE);
```

Comment:
@TomasVotruba @staabm I tried many things, and it keep crashing/infinite loop without this one left over. 

I think this part can be removed once we have all next/prev node attribute removed and removed the NodeConnectingVisitor, left with add SlimNodeConnectingVisitor for only lookup parent